modules/command/src/main/java/org/apache/fluo/command/FluoProgram.java

```diff
@@ -0,0 +1,125 @@
+    FluoExec fluoExec = new FluoExec();
+    FluoGetJars fluoGetJars = new FluoGetJars();
+    FluoInit fluoInit = new FluoInit();
+    FluoList fluoList = new FluoList();
+    FluoOracle fluoOracle = new FluoOracle();
+    FluoRemove fluoRemove = new FluoRemove();
+    FluoScan fluoScan = new FluoScan();
+    FluoStatus fluoStatus = new FluoStatus();
+    FluoWait fluoWait = new FluoWait();
+    FluoWorker fluoWorker = new FluoWorker();
+    JCommander jcommand = JCommander.newBuilder().addCommand(CONFIG.getCommandName(), fluoConfig)
+        .addCommand(EXEC.getCommandName(), fluoExec)
+        .addCommand(GET_JARS.getCommandName(), fluoGetJars)
+        .addCommand(INIT.getCommandName(), fluoInit).addCommand(LIST.getCommandName(), fluoList)
+        .addCommand(ORACLE.getCommandName(), fluoOracle)
+        .addCommand(REMOVE.getCommandName(), fluoRemove).addCommand(SCAN.getCommandName(), fluoScan)
+        .addCommand(STATUS.getCommandName(), fluoStatus).addCommand(WAIT.getCommandName(), fluoWait)
+        .addCommand(WORKER.getCommandName(), fluoWorker).build();
+
+    try {
+      jcommand.parse(args);
+    } catch (ParameterException e) {
+      System.err.println(e.getMessage());
+      String commandName = Optional.ofNullable(jcommand.getParsedCommand()).orElse("");
+      JCommander parsedCommandOrProgram =
+          Optional.ofNullable(jcommand.findCommandByAlias(commandName)).orElse(jcommand);
+      parsedCommandOrProgram.setProgramName(String.format("fluo %s", commandName));
+      parsedCommandOrProgram.usage();
+      System.exit(1);
+      return;
+    }
+
+    FluoCommandType parsedCommandType = FluoCommandType.valueOf(jcommand.getParsedCommand());
+
+    FluoCommand parsedCommand;
+    switch (parsedCommandType) {
+      case CONFIG:
+        parsedCommand = fluoConfig;
+        break;
+      case EXEC:
+        parsedCommand = fluoExec;
+        break;
+      case GET_JARS:
+        parsedCommand = fluoGetJars;
+        break;
+      case INIT:
+        parsedCommand = fluoInit;
+        break;
+      case LIST:
+        parsedCommand = fluoList;
+        break;
+      case ORACLE:
+        parsedCommand = fluoOracle;
+        break;
+      case REMOVE:
+        parsedCommand = fluoRemove;
+        break;
+      case SCAN:
+        parsedCommand = fluoScan;
+        break;
+      case STATUS:
+        parsedCommand = fluoStatus;
+        break;
+      case WAIT:
+        parsedCommand = fluoWait;
+        break;
+      case WORKER:
+        parsedCommand = fluoWorker;
+        break;
+      default:
+        System.err.println("Unrecognized command");
+        System.exit(1);
+        return;
+    }
+
+    String programName = String.format("fluo %s", parsedCommandType.getCommandName());
+    JCommander command = jcommand.findCommandByAlias(parsedCommandType.getCommandName());
+    command.setProgramName(programName);
+
+    if (parsedCommand.isHelp()) {
+      command.usage();
+      System.exit(1);
```

Comment:
Why exit(1) when someone runs help?

Reply:
I had `System.exit(1)` because that was the existing behavior from `BaseOpts` class. Though now that I'm looking at it we should probably just return here.